hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/queryplan/physical/impl/PhysicalOpBindJoinWithBoundJoin.java

```diff
@@ -0,0 +1,71 @@
+package se.liu.ida.hefquin.engine.queryplan.physical.impl;
+
+import se.liu.ida.hefquin.base.query.ExpectedVariables;
+import se.liu.ida.hefquin.base.query.SPARQLGraphPattern;
+import se.liu.ida.hefquin.engine.queryplan.executable.UnaryExecutableOp;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.ops.ExecOpBindJoinSPARQLwithBoundJoin;
+import se.liu.ida.hefquin.engine.queryplan.info.QueryPlanningInfo;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpGPAdd;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpGPOptAdd;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalPlanVisitor;
+import se.liu.ida.hefquin.federation.SPARQLEndpoint;
+
+/**
+ * A physical operator that implements (a batching version of) the bound-join
+ * algorithm that uses UNION clauses with variable renaming (as proposed in
+ * the FedX paper by Schwarte et al. 2011). The variable that is renamed can
+ * be any non-join variable.
+ *
+ * <p>
+ * <b>Semantics:</b> TODO
+ * </p>
```

Comment:
?

You can simply copy the corresponding part from the other `PhysicalOpBindJoinWith*` classes.